en/middleware/micrortps.md

```diff
@@ -132,10 +132,10 @@ rtps:
     send: true
 ```
 
-> **Note** The `px4_msgs` build process (only) runs the CMake macro `rosidl_generate_interfaces()` to generate ROS2 message header files, while `px4_ros_com` build process generates the IDL files through the `rosidl_generate_dds_interfaces()` CMake macro.
+> **Note** Currently, the `px4_msgs` build process (only) runs the CMake macro `rosidl_generate_interfaces()` to generate ROS2 message header and source files and also the required IDL files, but previously it was using the `rosidl_generate_dds_interfaces()` CMake macro for the same purpose. A change on the API between the ROS2 Crystal and Dashing versions required that the build process was also adjusted to generate different headers a source files according to the versions.
```

Comment:
There are some minor problems with text above. Is this still accurate:

> **Note** The `px4_msgs` build process (only) runs the CMake macro `rosidl_generate_interfaces()` to generate ROS2 message header and source files, and also the required IDL files (previously it used the `rosidl_generate_dds_interfaces()` CMake macro for the same purpose). This change was required because  different headers and source files are now needed for ROS2 Crystal and Dashing versions.

Reply:
> This change was required because different headers and source files are now needed for ROS2 Crystal and Dashing versions.

Well not exactly. The problem was that before Dashing (meaning Crystal and previous versions), the `rosidl_generate_interfaces()`, or generated IDL files which weren't compatible with the ones that `fastrtpsgen` required to generated the code for the agent (in Crystal), or they didn't generate any IDL files (which was the case of Ardent and Bouncy). So you required to use `rosidl_generate_dds_interfaces()` to generate the IDL files you required.
What happened is that an API change between ROS2 Crystal and ROS2 Dashing releases on the `rosidl_generate_dds_interfaces()` CMake module led to `rosidl_generate_dds_interfaces()` not generating IDL files from the injected msg files (it now rather expects that you feed a tuple composed by the path of the file and an IDL file that will be adjusted to the IDL file spec that `ros_idl` expects). On the other way, `rosidl_generate_interfaces()` now generates the IDL files you need to use on `fastrtpsgen` to generate the agent code.
TL;DR, what is required to say is that an API change led us to rely only in the `rosidl_generate_interfaces()` CMake module (used in `px4_msgs`) to generate the required IDL files we use on the microRTPS agent generation in `px4_ros_com`.

Reply:
Yeah, but you made me read it :-)

I've updated pretty much like you said in https://github.com/PX4/Devguide/pull/956/commits/7b4ab468e9f4318c355af6ac6e549ae0e4df8ba9 - OK?

Reply:
Yes that looks good!